File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojogroup.Specification;

import java.util.List;

public interface SpecificationService {

    /**
     * 带条件的规格分页查询
     *
     * @param tbSpecification
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbSpecification tbSpecification, int pageNum, int pageSize);

    /**
     * 批量删除规格
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 查询单个实体类
     *
     * @param id
     * @return
     */
    public Specification findOne(Long id);

    /**
     * 添加规格组（规格+规格项）
     *
     * @param specification
     */
    public void add(Specification specification);

    /**
     * 更新规格
     *
     * @param specification
     */
    public void update(Specification specification);

    /**
     * 返回全部的规格列表
     *
     * @return、】
     */
    public List<TbSpecification> findAll();
}

File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function ($http) {

    // 返回全部规格列表
    this.findAll = function () {
        return $http.get('../specification/findAll.do');
    };

    // 带条件分页查询规格
    this.search = function (page, rows, searchEntity) {
        return $http.post('../specification/search.do?page=' + page + '&rows=' + rows, searchEntity);
    };

    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id=' + id);
    };

    this.delete = function (ids) {
        return $http.get('../specification/delete.do?ids=' + ids);
    };

    this.add = function (entity) {
        return $http.post('../specification/add.do', entity);
    };

    this.update = function (entity) {
        return $http.post('../specification/update.do', entity);
    }
});